tests/xharness/TestLabel.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Mono.Unix.Native;
+
+#nullable enable
+
+namespace Xharness {
+	[AttributeUsage (AttributeTargets.Field)]
+	public class LabelAttribute : Attribute {
+		public string Label { get; }
+
+		public LabelAttribute (string label)
+		{
+			Label = label;
+		}
+	}
+
+	[Flags]
+	public enum TestLabel {
+		//{ "all", false },
```

Comment:
Does this do what you think it does? Traditionally, in C#, if you want these to act as flags, you need to give initializers with bit values.  For example, if I do this:
```
	[Flags]
	public enum Things {
		Red,
		Green,
		Blue,
	}

	class MainClass {
		public static void Main (string [] args)
		{
			Console.WriteLine ((int)Things.Red);
			Console.WriteLine ((int)Things.Green);
			Console.WriteLine ((int)Things.Blue);
			Console.WriteLine ((int)(Things.Red | Things.Green));
			Console.WriteLine (Things.Red | Things.Green);
		}
	}
```
The output is :
```
0
1
2
1
Green
```
But if I rewrite the enum like this:
```
	[Flags]
	public enum Things {
		Red = 1 << 0,
		Green = 1 << 1,
		Blue = 1 << 2,
	}
```
The output is:
```
1
2
4
3
Red, Green
```
which is what I think you would want.

In addition, you can do things like redefine `All` to be
```
All = Bcl | Mac | iOS, // etc,
```

Reply:
you are correct, I was thing adding a None and the do All as the oposite of None, which is probably simpler